string message_type
builtin_interfaces/Time sent_time
uint64 msg_num
byte[] serialized_msg
